powerline/segments/common/bat.py

```diff
@@ -62,19 +62,22 @@ def _fetch_battery_info(pl):
 				pl.debug('Not using DBUS+UPower as no batteries were found')
 
 	if os.path.isdir('/sys/class/power_supply'):
-		linux_bat_fmt = '/sys/class/power_supply/{0}/capacity'
+		_ac_powered = None
+		linux_supplier_fmt = '/sys/class/power_supply/{0}/capacity'
 		linux_ac_fmt = '/sys/class/power_supply/{0}/online'
-		for linux_bat in os.listdir('/sys/class/power_supply'):
-			cap_path = linux_bat_fmt.format(linux_bat)
-			online_path = linux_ac_fmt.format(linux_bat)
-			if linux_bat.startswith('BAT') and os.path.exists(cap_path):
-				pl.debug('Using /sys/class/power_supply with battery {0}', linux_bat)
+		for linux_supplier in os.listdir('/sys/class/power_supply'):
+			cap_path = linux_supplier_fmt.format(linux_supplier)
+			online_path = linux_ac_fmt.format(linux_supplier)
+			if linux_supplier.startswith('AC') and os.path.exists(online_path):
+				with open(online_path, 'r') as f:
+					_ac_powered = int(f.readline()) == 1
```

Comment:
This variable will be set only once if you leave it set here. You need to handle `online_path` in `_get_battery_status`, just add a case when it is `None` and handle it.